templates/corosync.conf.erb

```diff
@@ -7,18 +7,52 @@ totem {
 <% unless @cluster_name.nil? -%>
   cluster_name:                        <%= @cluster_name %>
 <% end -%>
+<% unless @token.nil? -%>
   token:                               <%= @token %>
+<% end -%>
+<% unless @token_retransmits_before_loss_const.nil? -%>
   token_retransmits_before_loss_const: <%= @token_retransmits_before_loss_const %>
+<% end -%>
+<% unless @join.nil? -%>
   join:                                <%= @join %>
-<% if @consensus -%>
+<% end -%>
+<% unless @consensus.nil? -%>
   consensus:                           <%= @consensus %>
 <% end -%>
-  vsftype:                             none
+<% unless @vsftype.nil? -%>
+  vsftype:                             <%= @vsftype %>
+<% end -%>
+<% unless @max_messages.nil? -%>
   max_messages:                        <%= @max_messages %>
-  clear_node_high_bit:                 yes
+<% end -%>
+<% unless @clear_node_high_bit.nil? -%>
+  clear_node_high_bit:                 <% @clear_node_high_bit %>
+<% end -%>
+<% unless @rrp_mode.nil? -%>
   rrp_mode:                            <%= @rrp_mode %>
+<% end -%>
   secauth:                             <%= @enable_secauth_real %>
+<% unless @threads.nil? -%>
   threads:                             <%= @threads %>
+<% end -%>
+<% if @unicast_addresses != 'UNSET' -%>
+  transport:                           udpu
+<% Array(@unicast_addresses.first).each_index do |interface| -%>
+  interface {
+<% Array(Array(@unicast_addresses.map{|x| Array(x).flatten}).transpose[interface]).each do |addr| -%>
```

Comment:
I'm not sure if we can write this smarter. I'm a bit scared to touch it. It worked in the initial udp conf file so it should be fine as is.